src/illuminatio/test_orchestrator.py

```diff
@@ -181,21 +184,28 @@ def _rewrite_ports_for_host(self, port_list, services_for_host):
             }
         rewritten_ports = {}
         wild_card_ports = {p for p in port_list if "*" in p}
-        numbered_ports = {p for p in port_list if "*" not in p}
+        numbered_ports = {p for p in port_list if is_numerical_port(p)}
+        # We assume that the other ports are named_ports
+        named_ports = set(port_list) - set(numbered_ports) - set(wild_card_ports)
         service_ports = [p for svc in services_for_host for p in svc.spec.ports]
         self.logger.debug("Svc ports: %s", service_ports)
+
         for wildcard_port in wild_card_ports:
             prefix = "-" if "-" in wildcard_port else ""
             # choose any port for wildcard
             rewritten_ports[wildcard_port] = "%s%s" % (
                 prefix,
                 str(service_ports[0].port),
             )
+
         for port in numbered_ports:
             prefix = "-" if "-" in port else ""
             port_int = int(port.replace("-", ""))
             service_ports_for_port = [
-                p for p in service_ports if p.target_port == port_int
+                p
+                for p in service_ports
+                # TODO Services could also contain named ports. We should cross-compare here (resvoledName == numerical)
```

Comment:
The comment is related to the following issue (I guess we think about different things):

We are iterating over `numbered_ports`, but it could be the case, that the NetworkPolicy itself contains a numerical port (e.g. 80), but the Service uses a name (e.g. http), which is defined within the container.
In this case we are fine, since there is a service for port 80, but the comparison (e.g. if 80 == http) would fail and let the case fail, hence we would need to resolve the service ports here first to handle each edge case correctly.

I decided to skip this and only leave a comment as this is a edge case. I guess someone will use named ports in the NetPol and Svc accordingly in general.